analysis/preprocess/Stage1_data_cleaning_gi_bleeds(vax).R

```diff
@@ -0,0 +1,390 @@
+                                 nrow(input))
+  
+  print('Inclusion criteria: Did not recieve a mixed vaccine products before 07-05-2021')
+  
+  # Trick to run the mixed vaccine code on dummy data with limited levels -> To ensure that the levels are the same in vax_cat_product variables
+  
+  input <- input %>%
+    mutate(AZ_date = ifelse(vax_date_AstraZeneca_1 < mixed_vax_threshold, 1,
+                            ifelse(vax_date_AstraZeneca_2 < mixed_vax_threshold, 1,
+                                   ifelse(vax_date_AstraZeneca_3 < mixed_vax_threshold, 1, 0)))) %>%
+    mutate(Moderna_date = ifelse(vax_date_Moderna_1 < mixed_vax_threshold, 1,
+                                 ifelse(vax_date_Moderna_2 < mixed_vax_threshold, 1,
+                                        ifelse(vax_date_Moderna_3 < mixed_vax_threshold, 1, 0)))) %>%
+    mutate(Pfizer_date = ifelse(vax_date_Pfizer_1 < mixed_vax_threshold, 1,
+                                ifelse(vax_date_Pfizer_2 < mixed_vax_threshold, 1,
+                                       ifelse(vax_date_Pfizer_3 < mixed_vax_threshold, 1, 0)))) %>%
+    rowwise() %>%
+    mutate(vax_mixed = sum(across(c("AZ_date", "Moderna_date", "Pfizer_date")), na.rm = T)) %>%
+    dplyr::filter(vax_mixed < 2)
+  
+  consort[nrow(consort)+1,] <- c("Inclusion criteria: Did not recieve a mixed vaccine products before 07-05-2021",
+                                 nrow(input))
+  
+  print('Inclusion criteria: Index date is before cohort end date')
+  
+  # Will remove anyone who is not fully vaccinated by the cohort end date
+  
+  input <- input %>% filter (!is.na(index_date) & index_date <= end_date_exposure & index_date >= start_date_delta)
+  consort[nrow(consort)+1,] <- c("Inclusion criteria: Index date is before cohort end date",
+                                 nrow(input))
+  
+} 
+
+#Apply outcome specific exclusions criteria
+#-------------------------------------------------#
+
+#Remove chronic people with Coeliac, IBD and Cirrhosis
+input <- input %>% 
+filter_at(vars(out_bin_crohn, out_bin_cirrhosis,out_bin_coeliac_disease), any_vars(.== TRUE))
+
+consort[nrow(consort)+1,] <- c("Inclusion criteria: Include people with chronic GI disease (IBD, Crhon and Coeliac)",
+                              nrow(input))
+    
+
+#Remove outcomes which aren't analysed
+input <- input %>% 
+  select(-c("out_date_ibs", "out_date_diarrhoea","out_date_nausea", "out_date_vomiting", "out_date_abdominal_paindiscomfort", "out_date_intestinal_obstruction","out_date_bowel_ischaemia",                
+            "out_date_belching", "out_date_abdominal_distension","out_date_bloody_stools","out_date_appendicitis","out_date_gallstones_disease", "out_date_nonalcoholic_steatohepatitis",
+            "out_date_acute_pancreatitis","out_date_gastro_oesophageal_reflux_disease","out_date_dyspepsia", "out_date_peptic_ulcer"))
+
+consort[nrow(consort)+1,] <- c("Exclusion criteria: Remove outcomes that aren't of interest",
```

Comment:
This is just removing columns which won't affect the numbers of rows (it is not an exclusion criteria)